src/pages/TokenDetails/index.tsx

```diff
@@ -79,10 +80,29 @@ export default function TokenDetails() {
   const location = useLocation()
   const { tokenAddress } = useParams<{ tokenAddress?: string }>()
   const token = useToken(tokenAddress)
+  const tokenWarning = tokenAddress ? checkWarning(tokenAddress) : null
+  const isBlockedToken = !!tokenWarning && !tokenWarning?.canProceed
+  const navigate = useNavigate()
+
+  const [continueSwap, setContinueSwap] = useState<{ resolve: (value: boolean | PromiseLike<boolean>) => void }>()
+  const shouldShowSpeedbump = !useIsUserAddedToken(token) && !!tokenWarning
```

Comment:
nit: I prefer `Boolean(tokenWarning)` (it's harder to misread or typo). You could also be more explicit and use `tokenWarning !== null`.